subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -314,46 +352,54 @@ static void send_pending_adv(struct k_work *work)
 		}
 	}
 
-	while ((adv = bt_mesh_adv_get_by_tag(ext_adv->tags, K_NO_WAIT))) {
-		/* busy == 0 means this was canceled */
-		if (!adv->ctx.busy) {
-			bt_mesh_adv_unref(adv);
-			continue;
+	while (1) {
```

Comment:
The problem with `while (1)` is that if this work runs on the sysworkq, then none of other works will be processed until mesh has actually managed to send  something.

Reply:
yeah, finally removed.